### PR TITLE
[vm_runtime] Implement new for native vector.

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

This is the first step for supporting native vectors. We will implement serializer and deserializer for native vectors in this PR first. In the PR to come, we will implement other apis like push/pop/get etc.

Some questions remained to be answered:
1. This diff is modifying parts of the value API due to the in place update nature of native functions. I don't see a clear way around this other than rethink our value api a bit cc @dariorussi 
2. What would the destroy API for vector be. IMHO we should be able to delete empty resource vector or an arbitrary value vector. Though I'm not so sure how to express that in Move right now. 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Added one test case

### CODE REVIEW

**Comment by vgao1996** [comment]
If I'm understanding correctly this is replacing #857 , right?

**Comment by runtian-zhou** [comment]
> If I'm understanding correctly this is replacing #857 , right?
Exactly. This PR is splitting #857 into two sections: this PR will implement Vector.new, serialize and deserialize of a native vector. I'll later put out another diff that implements push/pop/get etc.

**Comment by runtian-zhou** [comment]
Just did a major rebase to add some APIs for the updated value story

File: language/vm/vm_runtime/vm_runtime_types/src/value.rs [hunk 1]
```diff
@@ -626,6 +713,10 @@ impl GlobalRef {
         }
     }
 
+    pub fn get_root(&self) -> Rc<RefCell<RootAccessPath>> {
```
Comment (dariorussi) [inline language/vm/vm_runtime/vm_runtime_types/src/value.rs]: is this still needed?

**Pull request merged**